src/journal/journalctl.c

```diff
@@ -264,7 +264,9 @@ static int parse_boot_descriptor(const char *x, sd_id128_t *boot_id, int *offset
         sd_id128_t id = SD_ID128_NULL;
         int off = 0, r;
 
-        if (strlen(x) >= 32) {
+        if (streq(x, "all"))
+          return 1;
```

Comment:
indentation. please follow https://systemd.io/CODING_STYLE, i.e. 8ch indenting

Reply:
we usually follow the rule that functions shall never clobber output parameters on error, and always initialize them on success. Given that this is a success situation (i.e. it returns >= 0, rather than < 0) it hence should reset `*boot_id`, and `*offset` in that case.

If you do that, the calling side becomes a bit simpler too, as you don't have to initialize those two vars explicitly first.

Reply:
Also, I think it would be nicer to make the return value be == 0 in case of "all", and > 0 in case a normal boot descriptor is specified. That way the return value would map directly to arg_boot later on (except for the < 0 error case).